odoo/addons/test_new_api/models/test_new_api.py

```diff
@@ -1399,3 +1399,21 @@ class Prisoner(models.Model):
 
     name = fields.Char('Name')
     ship_ids = fields.Many2many('test_new_api.ship', 'test_new_api_crew', 'prisoner_id', 'ship_id')
+
+
+class Team(models.Model):
+    _name = 'test_new_api.team'
+    _description = 'Odoo Team'
+
+    name = fields.Char('Name')
+    parent_id = fields.Many2one('test_new_api.team', string='Parent Team')
+    member_ids = fields.One2many('test_new_api.team.member', 'team_id')
+
+
+class TeamMember(models.Model):
+    _name = 'test_new_api.team.member'
+    _description = 'Odoo Developer'
+
+    name = fields.Char('Name')
+    team_id = fields.Many2one('test_new_api.team')
+    parent_id = fields.Many2one('test_new_api.team', related='team_id.parent_id')
```

Comment:
Note that this is not exactly the case mentioned in the commit message, where it is actually a `_inherits`, where the behavior might differ? Maybe?
Or is it good to consider that it behaves kind of the same way?

Reply:
Good point.

I initially made the Member class without `parent_id` field in mind.
The I realized that It doesn't make sense for Member class to _inherits from Team class. That's why I ended up with a simple related field.

Let's wait for @odoo/rd-framework-py

Reply:
inheritance is mentioned to explain why `parent_id` may reference to another model